pkgs/cronet_http/example/lib/main.dart

```diff
@@ -8,21 +8,19 @@ import 'dart:io';
 import 'package:cronet_http/cronet_client.dart';
 import 'package:flutter/material.dart';
 import 'package:http/http.dart';
-import 'package:http/io_client.dart';
 
 import 'book.dart';
 
 void main() async {
-  late Client client;
+  Client? client;
   if (Platform.isAndroid) {
+    WidgetsFlutterBinding.ensureInitialized();
     client = CronetClient(await CronetEngine.build(
         cacheMode: CacheMode.memory, cacheMaxSize: 1024 * 1024));
-  } else {
-    client = IOClient();
```

Comment:
This is a great point! I think that my version is incorrect because it would not work if you wrote:

```dart
final c = Client();
c.get(...)
c.close();
final d = Client()
d.get(...)  # Exception because the client has been closed!
```

This formulation seems better.

```dart
void main() async {
  var client = Client.new; // The default Client.
  if (Platform.isAndroid) {
    WidgetsFlutterBinding.ensureInitialized();
    client = CronetClient.new;
  }
  runWithClient(() => runApp(const BookSearchApp()), client);
}
```